nomad/structs/structs.go

```diff
@@ -2174,6 +2174,10 @@ type Node struct {
 	// Raft Indexes
 	CreateIndex uint64
 	ModifyIndex uint64
+
+	// NodeMaxAllocs is only set if a NodeMaxAllocs value
+	// was set in the client config
+	NodeMaxAllocs *NodeMaxAllocs
```

Comment:
The only big design issue I have with this is keeping a struct here instead of having this be an integer field. We have a sensible zero-value (as you've identified in the struct you've made here), so let's hoist it up to this field. That way we don't have to do any of the nil pointer checks and get to simplify the config parsing bits because the config will map more directly to the node struct.

```suggestion
	// NodeMaxAllocs is only set if a NodeMaxAllocs value
	// was set in the client config
	NodeMaxAllocs int
```

(Ideally this field would be on the `Resources` struct for the node, but because we use that same struct for the allocation `Resources` and jobspect `Resources`, that's not a good idea. Kind of a shame in the design of those structs.)

As far as the type of the integer, you're going to be comparing it to a count and that count is going to be small (dozens not billions), so keeping it as an `int` is a little more ergonomic.